### ISSUE TITLE
Songs requested near the end of the night don't show until the queue is purged

### ISSUE THREAD

**Issue by kevinsaucier** [whitesoup12/remote-falcon#134]
No huge rush here, but wanted to document it.  This one is kind of a bug, as it is unexpected in the UI, even though the behavior makes sense if you know why.  I requested 'Baby Got Back' just before my show ended last night.  When BGB was done, the next song in the normal playlist played, then the playlist ended gracefully.  When I looked at the RF page today, BGB was missing from the list because I have it set to not add it back to the list until 5 more songs have played.  I checked the Viewer Queue in the Admin page and nothing was listed there, but clearing that queue caused BGB to reappear in the list on the RF viewer page.

So....an enhancement would be to show songs that are 'waiting to be re-added' in the Viewer Queue popup in the Admin page.  It would be ideal if you could X them from that list to make them reappear immediately, but the main part would be some indication as to why a song doesn't exist on the RF viewer page.   Also, having those songs reset when the entire queue is purged doesn't technically make sense, as they are no longer queued, but that could be a different 'fix'.  😁

**Comment by whitesoup12** [comment]
So I have both a workaround and a proposal. :)

Before turning on viewer control when your show starts, purge the queue. Then the counts will get reset and all sequences will appear.

As for the proposal, I like the idea of showing how many songs are left before a sequence will be visible again, but it makes more sense to have that on the Sequences page. Then there could be an option to reset all counts, or even just individual sequence counts. It wouldn't make sense to have them in the Jukebox Queue list because a hidden sequence was just played and would not be active in the queue.

**Comment by kevinsaucier** [comment]
Yup, forgot to mention the workaround.  🤦‍♂️  I do actually have a purge in the Lead In to my main playlist, so it would have been reset tonight.  I was working on my viewer page now, though, so it hadn't reset yet.  I just added the purge to my Lead Out as well so it will be purged when the show ends.    

There does still seem to be some oddity there, however, as even though BGB had reappeared, 'Christmas Cookies' was still missing until I purged the queue a second time.  I know those were the last two I requested last night so something may not be working exactly as expected there.  It's possible it had something to do with me added categories to those songs as well.  Not exactly sure.

LOL, I agree it doesn't make sense to have them be part of the Jukebox Queue, but that's what you tied it to, so I was trying to work with you!  🤣🤣

Yup, resetting them individually would be ideal but there is limited space on the Sequences page, and I'd still like to see that flag for when a sequence has an image associated so......  😁
